lib/fluent/plugin/out_redis.rb

```diff
@@ -47,22 +57,42 @@ def shutdown
     end
 
     def format(tag, time, record)
-      identifier = [tag, time].join(".")
-      [identifier, record].to_msgpack
+      record = inject_values_to_record(tag, time, record)
+      [tag, time, record].to_msgpack
+    end
+
+    def formatted_to_msgpack_binary
+      true
     end
 
     def write(chunk)
+      tag, time = expand_placeholders(chunk.metadata)
       @redis.pipelined {
-        chunk.open { |io|
-          begin
-            MessagePack::Unpacker.new(io).each.each_with_index { |record, index|
-              @redis.mapped_hmset "#{record[0]}.#{index}", record[1]
-            }
-          rescue EOFError
-            # EOFError always occured when reached end of chunk.
+        unless @allow_duplicate_key
+          chunk.open { |io|
+            begin
+              MessagePack::Unpacker.new(io).each.each_with_index { |record, index|
```

Comment:
Can we use `chunk.each` instead?
At least, use `.each.with_index` instead of .each.each_with_index`.